extensions/amp-story/1.0/amp-story.js

```diff
@@ -404,12 +404,20 @@ export class AmpStory extends AMP.BaseElement {
 
     if (isExperimentOn(this.win, 'amp-story-branching')) {
       this.registerAction('goToPage', invocation => {
-        const {args} = invocation;
+        const {args, caller} = invocation;
+
         if (args) {
           this.storeService_.dispatch(
               Action.SET_ADVANCEMENT_MODE, AdvancementMode.GO_TO_PAGE);
-          this.switchTo_(args['id'], NavigationDirection.NEXT);
+
+          if (caller['offsetParent'].tagName === 'AMP-SIDEBAR') {
+            this.sidebar_.getImpl()
+                .then(sidebarImpl => sidebarImpl.close_());
+            this.switchTo_(args['id'], NavigationDirection.NEXT);
```

Comment:
This is introducing a bug. Let's imagine this story has a third page:
1. Load the story, you're on the cover page
2. Open the sidebar and click on a link or action to go to page 2
3. You're on page2, you click next, you're on page 3
4. Navigate back
5. Expected: now on page2 / Actual: now on page1

(I haven't tried this exact scenario, but from the history state bug this should happen)

When the sidebar opens, it pushes a new history entry. Because you're using a promise to get the sidebar implementation, the `sidebarImpl.close()` will happen after the `switchTo`.
`switchTo` writes the history state in the entry added by the sidebar, that is then popped by `sidebarImpl.close()`. So the navigation path actually never got updated, resulting in the navigation bug described above.

An easy fix would be to call `switchTo` after the sidebar's history entry is popped.

Reply:
Synched offline but you're right that there was a historyState bug, but it actually corrects itself as you navigate forwards. 

With the changes I just pushed, when you navigate using a sidebar link (not action), checking the `ampStoryNavigationPath` will yield an undefined state, but that also corrects itself when you either navigates forwards or backwards (this is due to manipulating the history state to not show the fragment as the user is navigating via the sidebar links/actions).